devdoc/umock_lock_factory_pthread.md

```diff
@@ -0,0 +1,99 @@
+# umock_lock_factory_pthread
+
+## Overview
+
+`umock_lock_factory_pthread` is a module that implements the reader/writer lock factory for `umock_c` for the pthread library.
+
+## Exposed API
+
+```c
+    UMOCK_C_LOCK_HANDLE umock_lock_factory_create_lock(void* params);
+```
+
+## static lock functions
+
+```c
+static void umock_lock_pthread_acquire_shared(UMOCK_C_LOCK_HANDLE lock);
+static void umock_lock_pthread_release_shared(UMOCK_C_LOCK_HANDLE lock);
+static void umock_lock_pthread_acquire_exclusive(UMOCK_C_LOCK_HANDLE lock);
+static void umock_lock_pthread_release_exclusive(UMOCK_C_LOCK_HANDLE lock);
+static void umock_lock_pthread_release_destroy(UMOCK_C_LOCK_HANDLE lock);
```

Comment:
>release [](start = 31, length = 7)

destroy

Reply:
Fixed.

---
In reply to: [530647904](https://github.com/Azure/umock-c/pull/177#discussion_r530647904) [](ancestors = 530647904)